src/Chainweb/BlockHeaderDB/RestAPI/Server.hs

```diff
@@ -276,3 +295,35 @@ someBlockHeaderDbServer (SomeBlockHeaderDb (db :: BlockHeaderDb_ v c))
 someBlockHeaderDbServers :: ChainwebVersion -> [(ChainId, BlockHeaderDb)] -> SomeServer
 someBlockHeaderDbServers v = mconcat
     . fmap (someBlockHeaderDbServer . uncurry (someBlockHeaderDbVal v))
+
+-- -------------------------------------------------------------------------- --
+-- BlockHeader Event Stream
+
+someHeaderStreamServer :: PayloadCas cas => ChainwebVersion -> CutDb cas -> SomeServer
+someHeaderStreamServer (FromSing (SChainwebVersion :: Sing v)) cdb =
+    SomeServer (Proxy @(HeaderStreamApi v)) $ headerStreamServer cdb
+
+headerStreamServer
+    :: forall cas (v :: ChainwebVersionT)
+    .  PayloadCas cas
+    => CutDb cas
+    -> Server (HeaderStreamApi v)
+headerStreamServer cdb = headerStreamHandler cdb
+
+headerStreamHandler :: forall cas. PayloadCas cas => CutDb cas -> Tagged Handler Application
+headerStreamHandler db = Tagged $ \req respond -> do
+    chan <- newChan
+    void . SP.mapM_ (g >=> writeChan chan . f) . SP.concat $ blockDiffStream db
+    eventSourceAppChan chan req respond
+  where
+    cas :: PayloadDb cas
+    cas = _webBlockPayloadStoreCas $ view cutDbPayloadStore db
+
+    g :: BlockHeader -> IO HeaderUpdate
+    g bh = do
+        x <- casLookupM cas $ _blockPayloadHash bh
+        pure $ HeaderUpdate (ObjectEncoded bh) (length $ _payloadWithOutputsTransactions x)
```

Comment:
`Foldable.length` on `Vector` is safe, it's overridden to `V.length` which is `O(1)`.